### PR TITLE
[FrameworkBundle][ContainerLintCommand] Style messages

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

It gives a better feedback.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php [hunk 1]
```diff
@@ -47,15 +51,32 @@ protected function configure()
...
         $container->setParameter('container.build_hash', 'lint_container');
         $container->setParameter('container.build_time', time());
         $container->setParameter('container.build_id', 'lint_container');
 
         $container->addCompilerPass(new CheckTypeDeclarationsPass(true), PassConfig::TYPE_AFTER_REMOVING, -100);
 
-        $container->compile();
+        try {
+            $container->compile();
+        } catch (InvalidArgumentException $e) {
```
Comment (fancyweb) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]: Should we check if the exception was thrown in CheckTypeDeclarationsPass?

**Comment by fancyweb** [comment]
I rebased on master as this is a new behavior.

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php [hunk 2]
```diff
@@ -47,15 +51,32 @@ protected function configure()
...
+            $errorIo->error($e->getMessage());
+
+            return 1;
+        }
+
+        $errorIo->success('The injected values in the services seem to be valid.');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]: would this be a bit better?
`The container was lint successfully: all services are injected with values that are compatible with their type declarations.`

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php [hunk 3]
```diff
@@ -69,6 +90,10 @@ private function getContainerBuilder(): ContainerBuilder
         $kernel = $this->getApplication()->getKernel();
 
         if (!$kernel->isDebug() || !(new ConfigCache($kernel->getContainer()->getParameter('debug.container.dump'), true))->isFresh()) {
+            if (!$kernel instanceof Kernel) {
+                throw new RuntimeException('This command does not support the application kernel.');
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]: `This command does not support the application kernel: "%s" does not extend "%s".`
(same in similar messages)

Comment (fancyweb) [reply src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]: Done on 4.4, see https://github.com/symfony/symfony/pull/34985. I'm gonna wait for a merge on master to finish this one.

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php [hunk 4]
```diff
@@ -64,7 +65,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-        $container->compile();
+        try {
+            $container->compile();
+        } catch (InvalidArgumentException $e) {
+            $errorIo->error($e->getMessage());
+
+            return 1;
+        }
+
+        $errorIo->success('The container was lint successfully: all services are injected with values that are compatible with their type declarations.');
```
Comment (ro0NL) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]: shouldnt it be the regular $io?

**Review by ro0NL — COMMENTED** [review]
much appreciated :)

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php (hunk 4 above)
Comment (fancyweb) [reply src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]: You are right and we did that mistake in multiple places, I'm gonna fix them.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Pull request merged**